pymongo/client_session.py

```diff
@@ -368,6 +377,97 @@ def _inherit_option(self, name, val):
             return val
         return getattr(self.client, name)
 
+    def with_transaction(self, callback, read_concern=None, write_concern=None,
+                         read_preference=None):
+        """Execute a callback in a transaction.
+
+        .. code-block:: python
+
+          def callback(session):
+              orders = session.client.db.orders
+              inventory = session.client.db.inventory
+              orders.insert_one({"sku": "abc123", "qty": 100}, session=session)
+              inventory.update_one({"sku": "abc123", "qty": {"$gte": 100}},
+                                   {"$inc": {"qty": -100}}, session=session)
+
+          with client.start_session() as session:
+              session.with_transaction(callback)
+
+        In an ideal scenario, `with_transaction` will start a transaction,
+        execute the callback once, and commit the transaction. In the event of
+        error, the commit or entire transaction may need to be retried and
+        thus the callback may be invoked multiple times. Even though the
+        callback may be invoked multiple times, `with_transaction` ensures
+        that the transaction will be committed at most once on the server.
+
+        In order to safeguard applications from infinite retry loops,
+        `with_transaction` will cease retrying the callback and
+        `commit_transaction` if it has exceeded a fixed timeout period of 120
+        seconds. This limit is non-configurable and is intentionally twice the
```

Comment:
Nit: I am not sure pointing out the fact that this unconfigurable limit is 2x the default timeout adds any value here.

Reply:
Might be more useful to add this as a comment above the variable declaration for the max duration.